src/main/java/frc/robot/ChallengeFour.java

```diff
@@ -20,12 +20,15 @@
 
     //put constants here
     private static double FEEDER_SPEED = -0.4;
+    private static double INTAKE_SPEED = 0.85;
     private static double ZERO = 0;
 
     //put variables here
+    private double shooterSpeed = 0.5; // incremented as we move further back from target
 
     //this is the main controller class (which we have written before), which will call the update methods below. This is NOT an Xbox Controller
     private Controller controller;   
+    private double encoderDistance = controller.getDistanceTravelled("fL"); // change encoder position accordingly
```

Comment:
This value need to be constantly updated in the teleop loop. Just initialize it with 0, and then set its value each frame in the teleop loop.